src/terraform.yml

```diff
@@ -39,14 +39,16 @@ repository:
       - include: "#expressions"
 
   functions:
-    begin: (\w+)(\()
+    begin: (\w+|(?:provider::\w+::\w+))(\()
     name: meta.function-call.hcl
     comment: Built-in function calls
     beginCaptures:
       "1":
         patterns:
           - match: \b(abs|abspath|alltrue|anytrue|base64decode|base64encode|base64gzip|base64sha256|base64sha512|basename|bcrypt|can|ceil|chomp|chunklist|cidrhost|cidrnetmask|cidrsubnet|cidrsubnets|coalesce|coalescelist|compact|concat|contains|csvdecode|dirname|distinct|element|endswith|file|filebase64|filebase64sha256|filebase64sha512|fileexists|filemd5|fileset|filesha1|filesha256|filesha512|flatten|floor|format|formatdate|formatlist|indent|index|join|jsondecode|jsonencode|keys|length|log|lookup|lower|matchkeys|max|md5|merge|min|nonsensitive|one|parseint|pathexpand|plantimestamp|pow|range|regex|regexall|replace|reverse|rsadecrypt|sensitive|setintersection|setproduct|setsubtract|setunion|sha1|sha256|sha512|signum|slice|sort|split|startswith|strcontains|strrev|substr|sum|templatefile|textdecodebase64|textencodebase64|timeadd|timecmp|timestamp|title|tobool|tolist|tomap|tonumber|toset|tostring|transpose|trim|trimprefix|trimspace|trimsuffix|try|upper|urlencode|uuid|uuidv5|values|yamldecode|yamlencode|zipmap)\b
             name: support.function.builtin.terraform
+          - match: \bprovider::(?!null|false|true)[[:alpha:]][[:alnum:]_-]*::(?!null|false|true)[[:alpha:]][[:alnum:]_-]*\b
```

Comment:
I used the same regex that is used for hcl functions to identify valid identifiers using `:alpha:` and `:alnum:_-`. We could also look into simplifying this by using `\w` for example.

Reply:
Using `\w` sounds like a good idea to me